src/app/utils/query-url-sanitization.ts

```diff
@@ -1,67 +1,74 @@
+import { GRAPH_URL } from '../services/graph-constants';
 import { parseSampleUrl } from './sample-url-generation';
 
-// Matches patterns like "('<1f7ff346-c174-45e5-af38-294e51d9969a>')" or "('<key>')"
-const keyIdRegex = /\(\'\<?\{?[ ?0-9a-zA-Z-]*\}?\>?\'\)/g;
-
-// Matches patterns like "(query=<key>)" or "(itemat=<key>,mode=<mode>)"
-const functionParamInitialRegex = /(?<=\=).*?(?=,)/g;
-const functionParamFinalRegex = /(?<=\=).[^,]*(?=\))/g;
-
 // Matches pattterns within quotes e.g "displayName: Gupta"
 const quotedTextRegex = /"([^"]*)"/g;
 
-// Matches PII patterns
-const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
-const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
-const guidRegex =
- /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
+// matches strings that are all alphabets
+const allAlphaRegex = /^[A-Za-z]+$/;
 
+// matches strings with depracation identifier
+const depracationRegex = /_v2/gi;
 
 export function sanitizeQueryUrl(url: string): string {
   url = decodeURIComponent(url);
 
   // Extract query string
-  const { search } = parseSampleUrl(url);
+  const { search, queryVersion, requestUrl } = parseSampleUrl(url);
   const queryString: string = sanitizeQueryParameters(search);
 
-  // Drop query string
-  let sanitizedUrl = url.split('?')[0];
-
-  // Normalize parameters in param=<arbitraryKey> format with param={value} format
-  sanitizedUrl = sanitizedUrl.replace(functionParamInitialRegex, '<value>');
-  sanitizedUrl = sanitizedUrl.replace(functionParamFinalRegex, '<value>');
+  let resourceUrl = requestUrl;
+  const sections = requestUrl.split('/');
+  sections.forEach(segment => {
```

Comment:
please add a comment here that mentions
```
what is skipped in the sanitization process:
- NonIDs that are url segments from entities/entity sets/navigations from the metadata. They are expected to contain only alpha characters.
- NonIDs that indicate deprecation. It is of the form <another_non_id>_v2.

Remaining url segments are assumed to be IDs that need to be sanitized.
```
If there is a bug in this it will be easy for the person investigating this to understand the assumptions we are making here.